docs/docs/tutorials/debugging.md

```diff
@@ -0,0 +1,89 @@
+# :material-bug: Hardware Debugging
+
+## Background
+
+For debugging our boards, we use OpenOCD and GDB.
+
+[OpenOCD (Open On-Chip Debugger)](https://openocd.org/) is a tool that enables on-chip debugging, flash programming, and testing for microcontrollers. It will start a server that listens for commands from various sources such as GDB.
+
+[GDB (GNU Debugger)](https://www.sourceware.org/gdb/) is a debugging tool that enables inspection, modification, and control of program execution on microcontrollers.
+
+## Installation
+
+Use the following command to install OpenOCD:
+
+=== "Windows"
+    ```text
+    <!-- TODO: Rida documentation for OpenOCD -->
+    ```
+
+=== "Linux / Mac"
+    ```text
+    brew install openocd
+    ```
+
+It is likely the ARM GNU Toolchain containing GDB is already installed if you have followed the [Firmware Development Setup](../firmware/dev-setup.md) guide. Refer to this guide if it is not installed.
```

Comment:
Can you make the first bit of the sentence more direct about GDB, like "The ARM GNU Toolchain GDB program is already installed if you followed..."

Also "this guide" seems to refer to this debugging document, change to "that guide"